Week03/BusinessCard/app/src/main/java/tech/danielwaiguru/businesscard/MainActivity.kt

```diff
@@ -1,11 +1,95 @@
 package tech.danielwaiguru.businesscard
 
+import android.content.DialogInterface
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.animation.AnimationUtils
+import androidx.appcompat.app.AlertDialog
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+    private val androidTips = listOf(
+        "Lifecycle of Fragments",
+        "onAttach() - called first before onCreate()",
+        "onCreateView() - called when its time for the fragment to draw its UI for the first time",
+        "onViewCreated() - called after onCreateView(), here you can use findViewById()",
+        "onActivityCreated() - indicates that the activity's onCreate() has completed",
+        "onStart() - called once the fragments gets visible",
+        "onPause() - called by the system to indicate that the user is leaving the fragment",
+        "onStop() - stops the fragment",
+        "onDestroyView() - contains the logic UI specific things that need to be cleaned up",
+        "onDestroy() - does the final clean up of the fragment's state",
+        "onDetach() - notify that the fragment has been disassociated from its hosting activity"
+    )
+    companion object{
+        private const val TIP_KEY = "TIP KEY"
+    }
+    lateinit var currentTip: String
```

Comment:
pay attention to warnings in the IDE. Here it's advising you to scope this var as private

Reply:
Thank you 
i just resolved the crash by adding an else block as below 
``` 
currentTip = if (savedInstanceState != null){
            savedInstanceState.getString(TIP_KEY)!!
        } else {
            androidTips.random()
        }
        txtTips.text = currentTip
```